cdk/lib/payment-api.ts

```diff
@@ -1,22 +1,409 @@
+                : [emailSqsCodeArn.valueAsString],
+          }),
+          new GuPutCloudwatchMetricsPolicy(this),
+          new GuAllowPolicy(this, "AssumeOphanRole", {
+            actions: ["sts:AssumeRole"],
+            resources: [ophanRole.valueAsString],
+          }),
+          new GuAllowPolicy(this, "KinesisPut", {
+            actions: ["kinesis:*"],
+            resources: [kinesisStreamArn.valueAsString],
+          }),
+          new GuAllowPolicy(this, "SQSPut", {
+            actions: ["sqs:SendMessage"],
+            resources:
+              this.stage === "PROD"
+                ? [
+                    contributionsStoreSqsQueueCodeArn.valueAsString,
+                    contributionsStoreSqsQueueProdArn.valueAsString,
+                  ]
+                : [contributionsStoreSqsQueueCodeArn.valueAsString],
+          }),
+          new GuAllowPolicy(this, "SqsKmsEncryption", {
+            actions: ["kms:Encrypt"],
+            resources: [sqsKmsArn.valueAsString],
+          }),
+          new GuGetS3ObjectsPolicy(this, "SettingsBucket", {
+            bucketName: "support-admin-console",
+            paths: this.stage === "PROD" ? ["*/*"] : ["CODE/*"],
+          }),
+        ],
+      },
+    });
+
+    // TODO: remove this tag after the migration
+    const playAppAsg = playApp.autoScalingGroup;
+    Tags.of(playAppAsg).add("gu:riffraff:new-asg", "true");
+
+    // ---- Alarms ---- //
+
+    new GuAlarm(this, "NoHealthyInstancesAlarm", {
+      app,
+      alarmName: `[CDK] No healthy instances for ${app} in ${this.stage}`,
+      actionsEnabled: props.stage === "PROD",
+      threshold: 0.5,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HealthyHostCount",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: playApp.loadBalancer.loadBalancerFullName,
+          TargetGroup: playApp.targetGroup.targetGroupFullName,
+        },
+        statistic: "Average",
+        period: Duration.seconds(60),
+      }),
+      snsTopicName: "contributions-dev",
+    });
+
+    new GuAlarm(this, "High5XXRateAlarm", {
+      app,
+      alarmName: `[CDK] High 5XX rate for ${app} in ${this.stage}`,
+      actionsEnabled: props.stage === "PROD",
+      threshold: 3,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HTTPCode_Target_5XX_Count",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: playApp.loadBalancer.loadBalancerFullName,
+          TargetGroup: playApp.targetGroup.targetGroupFullName,
+        },
+        statistic: "Sum",
+        period: Duration.seconds(60),
+      }),
+      snsTopicName: "contributions-dev",
+    });
+
+    new GuAlarm(this, "NoPaypalPaymentsInTwoHours247Alarm", {
```

Comment:
An unfortunate side effect of dual stacking is that we will get all of these alarms twice. The original alarm in the cloudformation and the ones in CDK will mean we receive two emails every time an alarm goes off! I suppose that's good motivation to get phase 3 done!